milestones/api.py

```diff
@@ -163,7 +163,10 @@ def get_course_milestones(course_key, relationship=None):
     _validate_course_key(course_key)
 
     if relationship is not None:
-        _validate_milestone_relationship_type(relationship)
+        try:
+            _validate_milestone_relationship_type(relationship)
+        except exceptions.InvalidMilestoneRelationshipTypeException:
+            return []
```

Comment:
To be clear, I'm not suggesting we remove the relationship model, only the validation that occurs to make sure the relationship identifier passed to the milestones api exists in the database. We still need the relationship model to distinguish between the supported types of relationships.